[issue 5428] No longer let the worker store things in share/testresults

coolo opened the pull request:
The worker uploads everything through HTTP - not yet finished though and it goes together with the os-autoinst pull request.

Just for review/discussion for now

coolo commented:
that's rather sad, but more worker code == less coverage ;(

aaannz commented on lib/OpenQA/Controller/Running.pm:
```diff
@@ -39,11 +40,9 @@ sub init {
-    my $workerurl = $worker->{properties}->{WORKER_IP} . ':' . $workerport;
+    $self->stash('worker', $job->worker);
+    my $workerport = $job->worker->get_property('WORKER_PORT');
+    my $workerurl = $job->worker->get_property('WORKER_IP') . ':' . $workerport;
```
I think WORKER_PORT and IP aren't needed for anything anymore.

coolo replied:
depends on your definition of needing:

<pre>
./templates/test/running.html.ep:              <td>Worker: <%= $worker->{properties}->{WORKER_IP} %>:<%= $worker->{properties}->{WORKER_PORT} %></td>
</pre>

aaannz commented on lib/OpenQA/Schema/Result/JobModules.pm:
```diff
@@ -215,10 +182,11 @@ sub split_results($;$) {
     my $schema = OpenQA::Scheduler::schema();
     for my $tm (@{$results->{testmodules}}) {
-        my $r = _insert_tm($schema, $job, $tm);
+        my $r = $job->_insert_tm($schema, $tm);
```
That's a nitpick, but I think _ prefixed methods are usually meant as private. Since you edit the function anyway, can you rename it?

coolo replied:
yeah

aaannz commented on lib/OpenQA/Schema/Result/Workers.pm:
```diff
@@ -45,5 +46,47 @@ __PACKAGE__->has_many(properties => 'OpenQA::Schema::Result::WorkerProperties',
 # TODO
 # INSERT INTO workers (id, t_created) VALUES(0, datetime('now'));
 
+sub seen(;$) {
```
Are we going to continue to use function args prototyping even on methods? I think Perl::Critic complains on this (even on non-methods).

coolo replied:
hmm, what does it complain about exactly?

coolo replied:
found it - it wants us to remove the prototype because they have no influence. I consider it documentation still

coolo replied:
I added a .perlcriticrc to ignore this - it seems to be very controversial as I found several HOWTOs to ignore it :)

aaannz commented on lib/OpenQA/Worker/Jobs.pm:
```diff
@@ -163,6 +157,7 @@ sub stop_job($;$) {
     clean_pool();
     $job = undef;
     $worker = undef;
+    exit(0);
```
exit here? Did I miss something? Wouldn't this stop worker after each job?

coolo replied:
hmm, yeah - useful for debugging. Not for real :)

BTW: it would be great if the worker grabed a job directly after start and not wait for the timer to kick in.

aaannz commented:
Heh, I didn't notice your 'not yet finished' note, now those TODOs makes sense. :) 
I like the direction, but I feel I'll need a 'guided tour' or more time to comprehend how's going to work.

coolo commented:
The ultimative goal is to have openQA act on DB and not on files - so the worker reads the files and uploads them as needed. There is no need for us to store the things as os-autoinst spits them out. What matters is that we can dig them quickly in the webui.

See also https://progress.opensuse.org/issues/6496 for an idea how to make an even bigger mess with the files :)

coolo commented:
I rebased the branch and removed all debugging aids as it works pretty fine now.

What's missing:
 - writing a migration that splits all result.json files
 - upload the ulogs

coolo commented:
+ upload video and log file and ...

I will need a 2nd route for that - pngs as base64 JSON are already challenging :)

coolo commented:
wrote a migration now and rebased one more time :)

coolo commented:
I just counted: I added netto 0 TODOs as I removed 3 :)

left TODO:
 - fix interactive mode (I left that open intentionally as lnussel rewrites this anyway)
 - cleanup tmpdirs (can happen after merge to master as I'm not sure when to do it actually)
 - upload logs as already mentioned

coolo commented:
I'm done with my TODO

coolo commented:
I would like to merge this tomorrow morning and rebase Ludwig' changes together with him - so last chance for public humilation :)

nilxam commented:
Looks fine to me, I skipped DB schema update though.
